client/client.py

```diff
@@ -9,295 +9,375 @@
-		#STORE TERRAIN SCALE FOR LATER USE#
-		self.terrainScale = terrainClass.terrain.getRoot().getSz()
-	def setPlayerNum(self,int):
-		self.playernum = int
-		
-	def move(self, keyClass, terrainClass):
-		self.meTerrainHeight = terrainClass.terrain.getElevation(self.model.getX(),self.model.getY()) * self.terrainScale
-		self.camTerrainHeight = terrainClass.terrain.getElevation(camera.getX(),camera.getY()) * self.terrainScale
-		self.elapsed = globalClock.getDt()
-		base.camera.lookAt(self.actorHead)
-		if (keyClass.keyMap["left"]!=0):
-			self.model.setH(self.model.getH() + self.elapsed*300)
-			print str(self.model.getY()), str(self.model.getX())
-		if (keyClass.keyMap["right"]!=0):
-			self.model.setH(self.model.getH() - self.elapsed*300)
-		if (keyClass.keyMap["forward"]!=0):
-			self.model.setY(self.model, (self.elapsed*40))
-		if (keyClass.keyMap["back"]!=0):
-			self.model.setY(self.model, -(self.elapsed*40))
-		
-		if (keyClass.keyMap["forward"]!=0) or (keyClass.keyMap["left"]!=0) or (keyClass.keyMap["right"]!=0):
-			if self.isMoving is False:
-				self.model.loop("walk", fromFrame = 1, toFrame = 11)
-				self.isMoving = True
-		else:
-			if self.isMoving:
-				self.model.stop()
-				self.model.pose("walk",5)
-				self.isMoving = False
-		
-		self.model.setZ(self.meTerrainHeight)
-		
-		#CAMERA CONTROL#
-		self.camvec = self.model.getPos() - base.camera.getPos()
-		if (self.camTerrainHeight > self.meTerrainHeight):
-			camera.setZ(self.camTerrainHeight + 5)
-		else:
-			camera.setZ(self.meTerrainHeight + 5)
-		self.camvec.setZ(0)
-		self.camdist = self.camvec.length()
-		self.camvec.normalize()
-		if (self.camdist > 20):
-			base.camera.setPos(base.camera.getPos() + self.camvec*(self.camdist-20))
-			self.camdist = 20.0
-		if (self.camdist < 10):
-			base.camera.setPos(base.camera.getPos() - self.camvec*(10-self.camdist))
-			self.camdist = 10.0
-		return Task.cont
-		
+  def __init__(self, terrainClass):
+    self.model = Actor("models/ninja", {"walk":"models/ninja"})
+    self.actorHead = self.model.exposeJoint(None, 'modelRoot','Joint8')
+    #self.model.setScale(4)
+    self.playernum = None
+    self.timeSinceLastUpdate = 0
+    self.model.reparentTo(render)
+    self.model.setScale(0.5)
+    self.isMoving = False
+    self.AnimControl=self.model.getAnimControl('walk')
+    self.AnimControl.setPlayRate(0.05)
+    self.model.setBlend(frameBlend=1)
+    self.model.setPos(244,188,0)
+    #STORE TERRAIN SCALE FOR LATER USE#
+    self.terrainScale = terrainClass.terrain.getRoot().getSz()
+    base.camera.reparentTo(self.model)
+  def setPlayerNum(self,int):
+    self.playernum = int
+   
+  def move(self, keyClass, terrainClass):
+    #self.meTerrainHeight = terrainClass.terrain.getElevation(self.model.getX(),self.model.getY()) * self.terrainScale
+    #self.camTerrainHeight = terrainClass.terrain.getElevation(camera.getX(),camera.getY()) * self.terrainScale
+    self.camDummy = self.model.attachNewNode("camDummy")

```

Comment:
absolutely. It should be part of the Init process. That is a good thing you caught this as I was noticing a memory leak and that was probably causing it. I made the cam dummy for the camera to point at instead of the head because the head bobs up and down when animated causing the camera to shake up and down in a very unpleasant manner.